Tests/NIOPosixTests/AsyncChannelBootstrapTests.swift

```diff
@@ -656,6 +656,80 @@ final class AsyncChannelBootstrapTests: XCTestCase {
         }
     }
 
+    func testPipeBootstrap_whenInputNil() async throws {
+        let eventLoopGroup = MultiThreadedEventLoopGroup(numberOfThreads: 1)
+        let (pipe1ReadFH, pipe1WriteFH) = self.makePipeFileDescriptors()
+        let fromChannel = FileHandle(fileDescriptor: pipe1ReadFH, closeOnDealloc: false)
+        let channel: NIOAsyncChannel<ByteBuffer, ByteBuffer>
+
+        do {
+            channel = try await NIOPipeBootstrap(group: eventLoopGroup)
+                .takingOwnershipOfDescriptors(
+                    input: nil,
+                    output: pipe1WriteFH
+                ) { channel in
+                    channel.eventLoop.makeCompletedFuture {
+                        try NIOAsyncChannel(synchronouslyWrapping: channel)
+                    }
+                }
+        } catch {
+            [pipe1ReadFH, pipe1WriteFH].forEach { try? SystemCalls.close(descriptor: $0) }
+            throw error
+        }
+
+        var inboundIterator = channel.inbound.makeAsyncIterator()
+
+        do {
+            try await XCTAsyncAssertEqual(try await inboundIterator.next(), nil)
+
+            let response = ByteBuffer(string: "Response")
+            try await channel.outbound.write(response)
+            XCTAssertEqual(try fromChannel.readBytes(ofExactLength: response.readableBytes), Array(buffer: response))
```

Comment:
please use NIO to read out of this. This is the NIO test suite and we should use NIO to do the work :)

Reply:
Oh yes this is so easy now when supporting PipeBootstraps with only one side. I can just setup a PipeChannel for the other side as well :)